compiler/x/amd64/codegen/AMD64SystemLinkage.cpp

```diff
@@ -900,10 +900,10 @@ TR::AMD64ABILinkage::mapIncomingParms(
          {
          uint32_t align = getAlignment(parmCursor->getDataType());
          uint32_t alignMinus1 = (align <= AMD64_STACK_SLOT_SIZE) ? (AMD64_STACK_SLOT_SIZE - 1) : (align - 1);
-         uint32_t pos = -stackIndex;
-         pos += parmCursor->getSize();
+         uint32_t pos = (~stackIndex) + 1;
+         pos += static_cast<uint32_t>(parmCursor->getSize());
          pos = (pos + alignMinus1) & (~alignMinus1);
-         stackIndex = -pos;
+         stackIndex = (~pos) + 1;
          parmCursor->setParameterOffset(stackIndex);
```

Comment:
This also seems very dodgy. The `-stackIndex` and `-pos` parts. It seems we are negating unsigned values. Tagging @0xdaryl to review this one. Here was the warning:

```
warning C4146: unary minus operator applied to unsigned type, result still unsigned
```

Reply:
This code looks very odd. The `~ + 1` trick might silence the warning, but it doesn't clarify what the code is trying to do. My vote is to rewrite this code to be more sensible (or add a big comment block if this is actually the intended behaviour).

Reply:
I fixed the warning but I'm not sure if the code is correct, so wanted to point it out. I did not change the semantics of the code.